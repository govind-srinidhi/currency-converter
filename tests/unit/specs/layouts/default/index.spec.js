/**
 * @file It contains unit tests for default layout.
 */
 import { mount, createLocalVue } from "@vue/test-utils"
 import Vuetify from "vuetify"
 import Default from "@/layouts/default"
 import Vuex from "vuex"

 let wrapper

 const appBar                       = () => wrapper.findComponent({ ref: "app_bar_default" })
 const imageLogo                    = () => wrapper.findComponent({ ref: "image_logo" })
 const currencySnackbarNotification = () => wrapper.find("currency-snackbar-notification-stub")

 let store

 const state = {
  notification: undefined
 }

 const mutations = {
  setNotification: (state, notification) => {
    state.notification = notification
  }
 }

 const getters = {
  "shared/notification": state => state.notification,
 }

 const actions = {
  "shared/suppressNotification": jest.fn()
 }

 const mountDefault = () => {
   const localVue = createLocalVue()

   localVue.prototype.$CONSTANTS = {
    LABELS: {
      APP_NAME: "Currency converter"
    }
   }

   localVue.use(Vuetify)
   localVue.use(Vuex)

   store = new Vuex.Store({
    state,
    getters,
    mutations,
    actions
   })
 
   const vuetify = new Vuetify()
 
   wrapper = mount(Default, {
     sync: false,
     stubs: ["currency-snackbar-notification"],
     localVue,
     store,
     vuetify
   })
 }

 
 describe("default layout", () => {
   beforeEach(() => {
     mountDefault()
   })
 
   afterEach(() => {
     jest.clearAllMocks()
   })
 
   it("renders the notification page", () => {
     expect(wrapper).toBeTruthy()
   })
 
   it("renders elements correctly", async () => {
     expect(appBar().exists()).toBeTruthy()
     expect(imageLogo().exists()).toBeTruthy()
     expect(currencySnackbarNotification().exists()).toBeFalsy()
   })

   it("notification to be rendered, whenever notification is called", async () => {
     store.commit("setNotification", {
       type: "error",
       icon: "mdi-alert",
       text: "error notification"
     })
     await wrapper.vm.$nextTick()
     expect(currencySnackbarNotification().exists()).toBeTruthy()
     currencySnackbarNotification().vm.$emit("close")
     await wrapper.vm.$nextTick()
     expect(actions["shared/suppressNotification"]).toHaveBeenCalledTimes(1)
   })

   it("matches the default snapshot", async () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
 })