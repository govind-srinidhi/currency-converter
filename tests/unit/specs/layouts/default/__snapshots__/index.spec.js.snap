// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default layout matches the default snapshot 1`] = `
<div data-app="true" class="v-application v-application--is-ltr theme--light" id="app">
  <div class="v-application--wrap">
    <header class="v-sheet theme--light v-toolbar v-toolbar--flat v-app-bar v-app-bar--fixed primary lighten-3" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
      <div class="v-toolbar__content" style="height: 64px;">
        <div class="v-toolbar__title v-app-bar-title">
          <div class="v-app-bar-title__content">
            <div class="v-image v-responsive justify-left theme--light" style="height: 42px; width: 50px;">
              <transition-stub name="fade-transition" mode="in-out">
                <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
              </transition-stub>
              <div class="v-responsive__content"></div>
            </div>
          </div>
          <div class="v-app-bar-title__placeholder" style="visibility: hidden;">
            <div class="v-image v-responsive justify-left theme--light" style="height: 42px; width: 50px;">
              <transition-stub name="fade-transition" mode="in-out">
                <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
              </transition-stub>
              <div class="v-responsive__content"></div>
            </div>
          </div>
        </div><span class="subtitle-1">Currency converter</span>
      </div>
    </header>
    <main class="v-main" style="padding: 64px 0px 0px 0px;">
      <div class="v-main__wrap">
        <currency-snackbar-notification-stub pnotification="[object Object]"></currency-snackbar-notification-stub>
      </div>
    </main>
  </div>
</div>
`;
